### PR TITLE
Update ML Sample Pipeline to use ARM Service Connection

### LINKED ISSUE #36877: Refactor `trigger-ml-samples-pipeline` to utilize ARM connection
Versus static CS from kv. [reference location](https://github.com/Azure/azure-sdk-for-python/blob/3b012ac6e699596d407a1855e74ce4809c14d670/eng/pipelines/templates/jobs/trigger-ml-sample-pipeline.yml#L67)

This needs to be replaced with utilization of fed credential.

While this is being completed, we should also move the job from `eng/pipelines/templates/job` to `eng/pipelines/`. Wes would prefer top level jobs live there.

### PR DESCRIPTION
**Pull request by scbedd**
See title. resolves #36877

[Successful run with this PR's changes](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4052016&view=logs&j=41973b2a-913a-52a3-33b2-21df9bd9b31d&t=e85bff69-f1ff-5e07-09ca-d0a74d9eeaba)

@kdestin I'm moving the pipeline yml, but the [pipeline itself](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4052016&view=logs&j=41973b2a-913a-52a3-33b2-21df9bd9b31d&t=e85bff69-f1ff-5e07-09ca-d0a74d9eeaba) will remain in the same location triggered the same way.

### CODE REVIEW

File: eng/pipelines/trigger-ml-sample-pipeline.yml [hunk 1]
```diff
@@ -61,20 +61,20 @@ jobs:
...
+
+    - task: AzurePowerShell@5
+      displayName: Generate Samples Repo Changes
+      inputs:
+        azureSubscription: 'Azure SDK Engineering System'
+        azurePowerShellVersion: LatestVersion
+        pwsh: true
+        ScriptType: InlineScript
+        Inline: |
+          python scripts/devops_tasks/generate_ml_sample_update.py --ml-repo "$(mlrepo)" --ml-wheel-folder "$(Build.ArtifactStagingDirectory)/azure-ai-ml" --build-id "$(Build.BuildId)" --storage-account-name "$(StorageAccountName)";
```
Comment (weshaggard) [inline eng/pipelines/trigger-ml-sample-pipeline.yml:76]: Where is StorageAccountName set? Can it just be a variable in the yml?

File: eng/pipelines/trigger-ml-sample-pipeline.yml [hunk 2]
```diff
@@ -61,20 +61,20 @@ jobs:
...
-
     - pwsh: |
         cat "$(mlrepo)/sdk/python/setup.sh"
-      displayName: Display Updated setup.sh
+      displayName: Display setup.sh pre-update
+
+    - task: AzurePowerShell@5
+      displayName: Generate Samples Repo Changes
+      inputs:
+        azureSubscription: 'Azure SDK Engineering System'
```
Comment (weshaggard) [inline eng/pipelines/trigger-ml-sample-pipeline.yml:71]: If possible lets use `Azure SDK Artifacts` here.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/trigger-ml-sample-pipeline.yml (hunk 1 above)
Comment (scbedd) [reply eng/pipelines/trigger-ml-sample-pipeline.yml:76]: It's currently set a secret variable on the build def. It's just the name of the storage account so sure.

**Pull request merged**